android/app-example/src/main/java/com/badoo/ribs/example/rib/dialog_example/DialogExampleRouter.kt

```diff
@@ -7,31 +7,37 @@ import com.badoo.ribs.core.routing.action.RoutingAction
 import com.badoo.ribs.core.routing.action.RoutingAction.Companion.noop
 import com.badoo.ribs.dialog.DialogLauncher
 import com.badoo.ribs.example.rib.dialog_example.DialogExampleRouter.Configuration
+import com.badoo.ribs.example.rib.dialog_example.DialogExampleRouter.Configuration.Content
+import com.badoo.ribs.example.rib.dialog_example.DialogExampleRouter.Configuration.Overlay
 import com.badoo.ribs.example.rib.dialog_example.dialog.LazyDialog
-import com.badoo.ribs.example.rib.dialog_example.dialog.SimpleDialog
 import com.badoo.ribs.example.rib.dialog_example.dialog.RibDialog
+import com.badoo.ribs.example.rib.dialog_example.dialog.SimpleDialog
 import kotlinx.android.parcel.Parcelize
 
 class DialogExampleRouter(
     private val dialogLauncher: DialogLauncher,
     private val simpleDialog: SimpleDialog,
     private val lazyDialog: LazyDialog,
     private val ribDialog: RibDialog
-): Router<Configuration, DialogExampleView>(
-    initialConfiguration = Configuration.Default
+): Router<Configuration, Nothing, Content, Overlay, DialogExampleView>(
+    initialConfiguration = Content.Default
 ) {
     sealed class Configuration : Parcelable {
-        @Parcelize object Default : Configuration()
-        @Parcelize object SimpleDialog : Configuration(), Overlay
-        @Parcelize object LazyDialog : Configuration(), Overlay
-        @Parcelize object RibDialog : Configuration(), Overlay
+        sealed class Content : Configuration() {
```

Comment:
It could be like Configuration<Content, Overlay, Nothing>, which will reduce complexity on router and interactor interfaces

Reply:
Unfortunately can't see a way how type safety in `permanentParts: List<Permanent>` and `pushOverlay(overlay: Overlay)` can be enforced with this approach.

Reply:
Feel free to suggest later, now going with the current one.